### PR TITLE
Pipeline script for Hono nightly build

### PR DESCRIPTION
**Pull request by kaniyan**
Pipeline script for Hono nightly build

### CODE REVIEW

File: jenkins/Hono-Nightly-Pipeline.groovy [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ */
+def nightlyBuild() {
+    stage('Build') {
+        withMaven(maven: 'apache-maven-latest', jdk: 'jdk1.8.0-latest', options: [jacocoPublisher(disabled: true), artifactsPublisher(disabled: true)]) {
+            sh 'mvn clean package javadoc:aggregate'
+            sh 'mvn --projects :hono-service-auth,:hono-service-messaging,:hono-service-device-registry,:hono-adapter-http-vertx,:hono-adapter-mqtt-vertx,:hono-adapter-kura,:hono-example -am deploy -DcreateJavadoc=true -DenableEclipseJarSigner=true'
+        }
+    }
+}
```
Comment (sophokles73) [inline jenkins/Hono-Nightly-Pipeline.groovy]: mising EOL

File: jenkins/Hono-Nightly-Pipeline.groovy [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * Nightly build with maven (with jdk1.8.0-latest and apache-maven-latest as configured in 'Global Tool Configuration' in Jenkins).
+ *
+ */
+def nightlyBuild() {
+    stage('Build') {
+        withMaven(maven: 'apache-maven-latest', jdk: 'jdk1.8.0-latest', options: [jacocoPublisher(disabled: true), artifactsPublisher(disabled: true)]) {
+            sh 'mvn clean package javadoc:aggregate'
+            sh 'mvn --projects :hono-service-auth,:hono-service-messaging,:hono-service-device-registry,:hono-adapter-http-vertx,:hono-adapter-mqtt-vertx,:hono-adapter-kura,:hono-example -am deploy -DcreateJavadoc=true -DenableEclipseJarSigner=true'
```
Comment (sophokles73) [inline jenkins/Hono-Nightly-Pipeline.groovy]: I think we should also add the AMQP and the CoAP adapters here

Comment (kaniyan) [reply jenkins/Hono-Nightly-Pipeline.groovy]: Added :)

File: jenkins/Hono-Nightly-Pipeline.groovy (hunk 1 above)
Comment (kaniyan) [reply jenkins/Hono-Nightly-Pipeline.groovy]: Done :)

**Pull request merged**